src/awssdk_driver.rs

```diff
@@ -21,10 +19,35 @@ use aws_sdk_qldbsession::{
 use aws_smithy_client::bounds::SmithyConnector;
 use aws_smithy_client::erase::DynConnector;
 use aws_smithy_client::Client;
-use aws_types::app_name::AppName;
+use aws_smithy_http::{middleware::MapRequest, operation};
+use aws_smithy_http_tower::map_request::MapRequestLayer;
+use http::header::HeaderName;
+use http::Uri;
+use tower::ServiceBuilder;
 
 use crate::{error, settings::Environment};
 
+#[derive(Clone, Debug)]
+struct UserAgent;
+
+impl MapRequest for UserAgent {
+    type Error = Box<dyn Error + Send + Sync + 'static>;
+
+    fn apply(&self, request: operation::Request) -> Result<operation::Request, Self::Error> {
+        request.augment(|mut req, _conf| {
+            req.headers_mut().append(
+                HeaderName::from_static("x-amz-qldb-driver-version"),
+                amazon_qldb_driver::version().to_string().parse().unwrap(),
```

Comment:
I'm a little unsure what's going on here.

https://docs.rs/http/0.2.6/http/header/struct.HeaderMap.html#method.append says the args are generic over K and V. So, what is V? i.e. what is the `parse()` doing to the string?

It seems to me that we shouldn't be doing allocation (heap allocation for the string) plus parse work on every call. Furthermore, having `unwrap` makes me a little nervous as libraries should never panic as doing so can bring down a customer application.

Reply:
I see, so I think you're trying to use `FromStr` to get a `HeaderValue`. Is that right? If so, did you consider https://docs.rs/http/0.2.6/http/header/struct.HeaderValue.html#method.from_static which is a const fn?

Reply:
```suggestion
                HeaderValue::from_static(amazon_qldb_driver::version())
```